sunpy/visualization/animator/wcs.py

```diff
@@ -299,10 +303,30 @@ def plot_start_image_2d(self, ax):
 
         return im
 
+    def update_bounds(self):
+        """
+        Update vmin/vmax when set to 'auto'.
+        """
+        print(self.imshow_kwargs)
+        if 'vmax' in self.imshow_kwargs and self.imshow_kwargs['vmax'] == 'auto':
+            # set clip interval to 90% of maximum value
+            clip_interval = (1, 90)*u.percent
+            clip_percentages = clip_interval.to('%').value
+            vmin, vmax = AsymmetricPercentileInterval(*clip_percentages).get_limits(self.data_transposed)
+            return vmin, vmax
+        else:
+            return None
+
     def update_plot_2d(self, val, im, slider):
         """
         Update the image plot.
         """
         self.axes.reset_wcs(wcs=self.wcs, slices=self.slices_wcsaxes)
         im.set_array(self.data_transposed)
+
+        bounds = self.update_bounds()
+        if bounds is not None:
+            vmin, vmax = bounds
+            im.set_clim(vmin, vmax)
```

Comment:
Could the function itself not update the im instead of calling it and then checking if its none?

Reply:
This would be an issue because the initial plots are updated by the vmin/vmax passed to it. So hence I did it this way

Reply:
Also, the data to be displayed (2D plot is being updated here, so I thought updating the data and the vmin/vmax together would work better.